.github/workflows/refresh-lockfiles.yml

```diff
@@ -0,0 +1,111 @@
+# This workflow periodically creates new environment lock files based on the newest
+# available packages and dependencies.
+#
+# Environment specifications are given as conda environment.yml files found in
+# `requirements/ci/py**.yml`.  These state the packages required, the conda channels
+# that the packages will be pulled from, and any versions of packages that need to be
+# pinned at specific versions.
+#
+# For environments that have changed, a pull request will be made and submitted
+# to the main branch
+
+name: Refresh Lockfiles
+
+
+on:
+  workflow_dispatch:
+  workflow_call:
+
+
+jobs:
+
+  get_python_matrix:
+    # Determines which Python versions should be included in the matrix used in
+    # the gen_lockfiles job.
+    if: "github.repository_owner == 'SciTools'"
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.get_py.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v3
+      - id: get_py
+        run: echo "::set-output name=matrix::$(ls -1 requirements/ci/py*.yml | xargs -n1 basename | sed 's/....$//' | jq -cnR '[inputs]')"
```

Comment:
Note this does fix us to ensuring that the `requirements` are stored in the same location in each repo and follow a similar name formatting. If you don't like this we could change this to be an input to the workflow. But I kept it as I though it would be helpful to encourage consistency across the repos, after all I don't see why we need different locations for requirements

Reply:
I adopted a different pattern in `geovista` using one YAML file and injecting the python version at runtime (via `tox`).

I've found that the only thing that tends to change over time in the YAML file is the version of `python`.

Anyways, at the moment I think we should definitely just stick to the existing pattern, and possibly consider a different strategy at a later date :+1:

Reply:
Absolutely! I did notice we also do that in cf-units (https://github.com/SciTools/cf-units/blob/main/tox.ini#L31) so that is definitely the goal of where we want to be for all our repos requirements management, but we just need to wait until we adopt tox for everything.